libraries/AP_Motors/AP_Motors_Class.h

```diff
@@ -349,7 +349,20 @@ class AP_Motors {
     float _throttle_radio_passthrough; // throttle/collective input from pilot in 0 ~ 1 range.  used for setup and providing servo feedback while landed
     float _yaw_radio_passthrough;      // yaw input from pilot in -1 ~ +1 range.  used for setup and providing servo feedback while landed
 
-    AP_Int8             _pwm_type;            // PWM output type
+    enum class PWMType : uint8_t {
+        NORMAL     = 0,
+        ONESHOT    = 1,
+        ONESHOT125 = 2,
+        BRUSHED    = 3,
+        DSHOT150   = 4,
+        DSHOT300   = 5,
+        DSHOT600   = 6,
```

Comment:
https://github.com/ArduPilot/ardupilot/blob/3a834e83c77b47bab505bec888b76dc239c75386/Tools/AP_Periph/rc_out.cpp#L74

which uses https://github.com/ArduPilot/ardupilot/blob/3a834e83c77b47bab505bec888b76dc239c75386/libraries/AP_HAL/RCOutput.h#L211

Reply:
Periph is the same as Copter.

... so this is all garbage, both in the MSP protocol and in ArduPilot's filling of the field.

We should unify it all (based on an enumeration in AP_HAL::RCOut`?), but that's going to be a lot of hairy parameter conversion.  I can't help but think we're mixing different concepts in here somewhere...

Reply:
Periph is one more than copter because of the NONE at the beginning. I think both use the same eventually but periph exposes it directly to the user and copter has a +1 somewhere.